lib/pipeline.js

```diff
@@ -291,43 +332,68 @@ class PipelineModel extends BaseModel {
                     .then((existingJobs) => {
                         const jobsToSync = [];
                         const jobsProcessed = [];
+                        const triggersToSync = [];
                         const parsedConfigJobNames = Object.keys(parsedConfig.jobs);
+                        const pipelineId = this.id;
+                        let requiresList = [];
 
                         // Loop through all existing jobs
                         existingJobs.forEach((job) => {
+                            const jobName = job.name;
+
                             // if it's in the yaml, update it
-                            if (parsedConfigJobNames.includes(job.name)) {
-                                job.permutations = parsedConfig.jobs[job.name];
+                            if (parsedConfigJobNames.includes(jobName)) {
+                                const permutations = parsedConfig.jobs[jobName];
+
+                                requiresList = permutations[0].requires || [];
+                                job.permutations = permutations;
                                 job.archived = false;
                                 jobsToSync.push(job.update());
-                            // if it's not in the yaml and archive it
+                            // if it's not in the yaml then archive it
                             } else if (!job.isPR()) {
                                 job.archived = true;
                                 jobsToSync.push(job.update());
                             }
+
```

Comment:
Should also clean up the triggers table for jobs that get archived. E.g. I previously have a job `A` requries `[~sd@123:main]`. After I removed job A from my yaml, there shouldn't be a trigger record for `{ src: ~sd@123:main, dest: ~sd@456:A }`